### PR TITLE
Ensure that required git version is used

### LINKED ISSUE #774: Warn on old versions of git
Lets avoid issues like #744. I think 2.3.x is the current minimum for everything to work as expected.

### PR DESCRIPTION
**Pull request by ikesyo**
Resolves #774. How do you feel about the implementation?

### CODE REVIEW

File: Source/carthage/main.swift [hunk 1]
```diff
@@ -9,8 +9,14 @@
 import CarthageKit
 import Commandant
 import Foundation
-import Result
+import ReactiveCocoa
 import ReactiveTask
+import Result
+
+guard ensureGitVersion().first()?.value == true else {
+	fputs("Carthage requires git \(CarthageRequiredGitVersion) or later. Please update your git version at first.\n", stderr)
```
Comment (mdiep) [inline Source/carthage/main.swift]: Can we drop the second sentence and just make it `Carthage requires git \(CarthageRequiredGitVersion) or later.`?

File: Source/CarthageKit/Git.swift [hunk 2]
```diff
@@ -11,6 +11,9 @@ import Result
 import ReactiveCocoa
 import ReactiveTask
 
+/// The git version Carthage requires at least.
+public let CarthageRequiredGitVersion = "2.3.0"
```
Comment (NachoSoto) [inline Source/CarthageKit/Git.swift]: Awesome

**Comment by NachoSoto** [comment]
:+1:

**Comment by mdiep** [comment]
This is really great. :sparkles:

File: Source/carthage/main.swift (hunk 1 above)
Comment (ikesyo) [inline Source/carthage/main.swift]: Fixed in 74ba78b. :zap:  `\n` is still needed as we already do [here](https://github.com/Carthage/Carthage/blob/ensure-git-version/Source/carthage/main.swift#L35).

**Comment by ikesyo** [comment]
Thank you for the reviews! :v:

**Pull request merged**